data-serving/data-service/test/controllers/preprocessor.test.ts

```diff
@@ -0,0 +1,221 @@
+import { Request, Response } from 'express';
+
+import { Case } from '../../src/model/case';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import app from '../../src/index';
+import minimalCase from './../model/data/case.minimal.json';
+import { setRevisionMetadata } from '../../src/controllers/preprocessor';
+import supertest from 'supertest';
+
+let mongoServer: MongoMemoryServer;
+let dateNowSpy: any;
+
+beforeAll(() => {
+    mongoServer = new MongoMemoryServer();
+    dateNowSpy = jest
+        .spyOn(Date, 'now')
+        .mockImplementation(() => Date.parse('2020-03-03'));
+});
+
+beforeEach(() => {
+    supertest.agent(app);
+    return Case.deleteMany({});
+});
+
+afterAll(() => {
+    dateNowSpy.mockRestore();
+    return mongoServer.stop();
+});
+
+describe('create', () => {
+    it('create sets create metadata', async () => {
```

Comment:
Super great tests, one hyper-nit: don't need to repeat expressions from the `describe` block in `it` blocks (I think it's preferred for them to [read fluently](https://jestjs.io/docs/en/api#describename-fn)).

Reply:
oops totally right, thanks!